itests/niporep_manual_test.go

```diff
@@ -29,129 +29,328 @@ import (
-			minerB.AssertNoPower(ctx)
+			// Our test miners should have no power as they have yet to onboard and activate any sectors
+			for _, miner := range miners {
+				miner.AssertNoPower()
+			}
 
-			// Verify that ProveCommitSectorsNI rejects messages with invalid parameters
-			verifyProveCommitSectorsNIErrorConditions(ctx, t, minerB, sealProofType)
-
-			// ---- Miner B onboards a CC sector
-			var bSectorNum abi.SectorNumber
-			var bRespCh chan kit.WindowPostResp
-			var bWdPostCancelF context.CancelFunc
-
-			// Onboard a CC sector with Miner B using NI-PoRep
-			bSectorNum, bRespCh, bWdPostCancelF = minerB.OnboardCCSector(ctx, sealProofType)
-			// Miner B should still not have power as power can only be gained after sector is activated i.e. the first WindowPost is submitted for it
-			minerB.AssertNoPower(ctx)
-
-			// Check that the sector-activated event was emitted
-			{
-				expectedEntries := []types.EventEntry{
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "$type", Value: must.One(ipld.Encode(basicnode.NewString("sector-activated"), dagcbor.Encode))},
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "sector", Value: must.One(ipld.Encode(basicnode.NewInt(int64(bSectorNum)), dagcbor.Encode))},
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "unsealed-cid", Value: must.One(ipld.Encode(datamodel.Null, dagcbor.Encode))},
+			sectors := make([][]abi.SectorNumber, len(tc.miners))
+
+			for i, tcMiner := range tc.miners {
+				miner := miners[i]
+				var expectSuccesses int
+				for _, ec := range tcMiner.sectorsToOnboard {
+					if ec.IsSuccess() {
+						expectSuccesses++
+					}
 				}
-				from := head.Height()
-				recentEvents, err := client.FullNode.GetActorEventsRaw(ctx, &types.ActorEventFilter{FromHeight: &from})
+
+				head, err = client.ChainHead(ctx)
 				req.NoError(err)
-				req.Len(recentEvents, 1)
-				req.Equal(expectedEntries, recentEvents[0].Entries)
-			}
 
-			// Ensure that the block miner checks for and waits for posts during the appropriate proving window from our new miner with a sector
-			blockMiner.WatchMinerForPost(minerB.ActorAddr)
+				// Onboard CC sectors to this test miner using NI-PoRep
+				sectors[i] = miner.OnboardSectors(
+					sealProofType,
+					false,
+					len(tcMiner.sectorsToOnboard),
+					kit.WithExpectedExitCodes(tcMiner.sectorsToOnboard),
+					kit.WithRequireActivationSuccess(tcMiner.allOrNothing),
+					kit.WithModifyNIActivationsBeforeSubmit(func(activations []miner14.SectorNIActivationInfo) []miner14.SectorNIActivationInfo {
+						head, err := client.ChainHead(ctx)
+						req.NoError(err)
+
+						for j, ec := range tcMiner.sectorsToOnboard {
+							if !ec.IsSuccess() {
+								// Set the expiration in the past to ensure the sector fails onboarding
+								activations[j].Expiration = head.Height() - 100
+							}
+						}
+						return activations
+					}),
+				)
+
+				req.Len(sectors[i], expectSuccesses)
```

Comment:
The `sectors[i]` returned by `OnboardSectors` really contains exit codes for all sectors passed to `OnboardSectors`. So, I am not sure how these can always be equal.

Reply:
it only contains sector numbers now for sectors that returned `exitcode.Ok`, so we should only get the ones successfully onboarded